### PR TITLE
🌱 cron: structured logging

### PR DESCRIPTION
**Pull request by raghavkaul**
#### What kind of change does this PR introduce?
Clean up cron logging so that log message severity level can be parsed by stackdriver, and so that messages aren't automatically marked at the error level.

### CODE REVIEW

File: log/log.go [hunk 1]
```diff
@@ -41,6 +42,23 @@ func NewLogger(logLevel Level) *Logger {
...
 }
 
+// NewCronLogger creates an instance of *Logger.
+func NewCronLogger(logLevel Level) *Logger {
+	logrusLog := logrus.New()
+
+	// Configure logger
+	// Don't log to stderr by default (stackdriver treats stderr as error severity)
+	logrusLog.SetOutput(os.Stdout)
+	// for stackdriver, see: https://cloud.google.com/stackdriver/docs/solutions/gke/managing-logs
```
Comment (spencerschrock) [inline log/log.go:52]: which part of the link explains this, is it "Structured logging" ?

File: log/log.go [hunk 2]
```diff
@@ -41,6 +42,23 @@ func NewLogger(logLevel Level) *Logger {
...
+	logrusLog.SetFormatter(&logrus.JSONFormatter{})
```
Comment (spencerschrock) [inline log/log.go:53]: do we need to customize anything by providing options?

i.e mapping log level to `severity`?
> Logs generated with glog will have the common fields parsed, for example, `severity`, `pid`, `source_file`, `source_line`

**Review by spencerschrock — COMMENTED** [review]
@calebbrown FYI

File: log/log.go [hunk 3]
```diff
@@ -41,6 +42,23 @@ func NewLogger(logLevel Level) *Logger {
 	return NewLogrusLogger(logrusLog)
 }
 
+// NewCronLogger creates an instance of *Logger.
+func NewCronLogger(logLevel Level) *Logger {
+	logrusLog := logrus.New()
```
[the remark this answers is not in the document]
Comment (raghavkaul) [reply log/log.go:47]: Sure, we could pass a flag to `NewLogger`, something like `structured bool`? or maybe`toStackdriver bool`

[the remark this answers is not in the document]
Comment (spencerschrock) [reply log/log.go:47]: I meant more along the lines of:

```go
// NewCronLogger creates an instance of *Logger.
func NewCronLogger(logLevel Level) *Logger {
	logrusLog := NewLogger(logLevel)

	// Configure logger
	// Don't log to stderr by default (stackdriver treats stderr as error severity)
	logrusLog.SetOutput(os.Stdout)
	// for stackdriver, see: https://cloud.google.com/stackdriver/docs/solutions/gke/managing-logs
	logrusLog.SetFormatter(&logrus.JSONFormatter{})

	return NewLogrusLogger(logrusLog)
}
```

[the remark this answers is not in the document]
Comment (raghavkaul) [reply log/log.go:47]: I believe `NewLogger` returns an interface that conforms with [`go-logr`](https://github.com/go-logr/logr), but we need to set logrus-specific settings, so I think we'd need to re-implement `NewLogger`, or change its signature

[the remark this answers is not in the document]
Comment (spencerschrock) [reply log/log.go:47]: Ah, feel free to ignore then.

File: log/log.go [hunk 4]
```diff
@@ -41,6 +42,26 @@ func NewLogger(logLevel Level) *Logger {
...
+
+	// Configure logger
+	// Don't log to stderr by default (stackdriver treats stderr as error severity)
+	logrusLog.SetOutput(os.Stdout)
```
Comment (spencerschrock) [inline log/log.go:51]: Is this still true now that you specify the severity field below?
https://stackoverflow.com/a/44187839

**Review by spencerschrock — APPROVED** [review]

File: log/log.go (hunk 4 above)
Comment (raghavkaul) [reply log/log.go]: Makes sense to me to remove.

**Pull request merged**